src/parser/jobs/gnb/modules/Ammo.tsx

```diff
@@ -42,371 +19,93 @@ const LEFTOVER_AMMO_SEVERITY_TIERS = {
-	private totalGeneratedAmmo = 0 // Keep track of the total amount of generated ammo over the fight
-	private erroneousCircles = 0 // This is my new NEW band name.
 
-	@dependency private checklist!: Checklist
 	@dependency private suggestions!: Suggestions
+	@dependency private checklist!: Checklist
 
-	protected override init() {
-		this.addEventHook('init', this.pushToHistory)
-		this.addEventHook(
-			'cast',
-			{
-				by: 'player',
-				abilityId: Object.keys(ON_CAST_GENERATORS).map(Number),
-			},
-			this.onCastGenerator,
-		)
-		this.addEventHook(
-			'combo',
-			{
-				by: 'player',
-				abilityId: Object.keys(ON_COMBO_GENERATORS).map(Number),
-			},
-			this.onComboGenerator,
-		)
-		this.addEventHook(
-			'cast',
-			{
-				by: 'player',
-				abilityId: Object.keys(AMMO_SPENDERS).map(Number),
-			},
-			this.onSpender,
-		)
-		this.addEventHook('normaliseddamage', {by: 'player', abilityId: ACTIONS.FATED_CIRCLE.id}, this.onFatedCircle)
-		this.addEventHook('death', {to: 'player'}, this.onDeath)
-		this.addEventHook('complete', this.onComplete)
-	}
-
-	private onFatedCircle(event: NormalisedDamageEvent) {
-		if (event.hitCount < 2) {
-			this.erroneousCircles++
-		}
-	}
-
-	private onCastGenerator(event: CastEvent) {
-		const abilityId = event.ability.guid
-		const generatedAmmo = ON_CAST_GENERATORS[abilityId]
-		this.currentAbility = event.ability.name
-
-		this.addGeneratedAmmoAndPush(generatedAmmo, abilityId)
-	}
-
-	private onComboGenerator(event: LegacyComboEvent) {
-		const abilityId = event.ability.guid
-		const generatedAmmo = ON_COMBO_GENERATORS[abilityId]
-		this.currentAbility = event.ability.name
-
-		this.addGeneratedAmmoAndPush(generatedAmmo, abilityId)
-	}
-
-	private addGeneratedAmmoAndPush(generatedAmmo: number, abilityId: number) {
-		const originalAmmo = this.ammo
-		this.ammo += generatedAmmo
-		this.totalGeneratedAmmo += generatedAmmo
-		if (this.ammo > MAX_AMMO) {
-			const waste = this.ammo - MAX_AMMO
-			this.wasteBySource[abilityId] += waste
-			this.ammo = MAX_AMMO
-
-			this.pushToOvercapHistory(originalAmmo, 0, OvercapTiming.BEFORE)
-			this.pushToOvercapHistory(originalAmmo + generatedAmmo, waste, OvercapTiming.AFTER)
-		}
-
-		this.currentAbilityImpact = ActionImpact.GENERATOR
-		this.pushToHistory()
-	}
-
-	private onSpender(event: CastEvent) {
-		this.ammo = this.ammo - AMMO_SPENDERS[event.ability.guid]
-		this.currentAbility = event.ability.name
-		this.currentAbilityImpact = ActionImpact.SPENDER
-		this.pushToHistory()
-	}
-
-	private onDeath() {
-		this.wasteBySource[ACTIONS.RAISE.id] += this.ammo
-		this.dumpRemainingResources()
-	}
+	private ammoGauge = this.add(new CounterGauge({
+		maximum: MAX_AMMO,
+		/* We'll see how people like it being in the timeline over a seperate chart.
+		I've grown to prefer it on the timeline myself
+		chart: {label: 'Ammo', color: JOBS.GUNBREAKER.colour}, */
+		graph: {
+			handle: 'ammo',
+			label: <Trans id="gnb.gauge.resource.ammoLabel"> Ammo </Trans>,
```

Comment:
Prob don't need these extra spaces if it's going in the timeline, worth checking but nbd.